### DISCUSSION TITLE
Support Ability Challenge

### DISCUSSION THREAD

**Discussion by VVilmot** [Albeoris/Memoria#172] [Ideas]
Hi there,

I had an idea for a challenge where you automatically equip/unequip the Support Abilities on characters equipment. Combined with No AP gain and an ability randomiser, it could be very tactical what you use. 

After much digging through the original Assembly-CSharp.dll file, I found a line of code that can do this.

{ } > EquipUI > UpdateCharacterSA

int num = (int)ff9ITEM_DATA.ability[j];
if (num != 0 && 192 <= num)
{
	list.Add(num - 192);
}

Add a line of code here, to become;

int num = (int)ff9ITEM_DATA.ability[j];
if (num != 0 && 192 <= num)
{
	list.Add(num - 192);
	ff9abil.FF9Abil_SetEnableSA((int)player.info.slot_no, num, true);       // new line of code
}

It will now automatically equip Support Abilities (regardless of stones), but will only automatically unequip them if they are not learnt.

It might be a nice challenge for people to toggle on/off in Memoria.

**Comment by Tirlititi** [comment]
I don't remember how you did for your randomizer but that feature is now a part of Memoria with the commit a21e372
